### PR TITLE
[WASI-NN] mlx: Lint codes and refine cmake.

### PR DESCRIPTION
**Pull request by q82419**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: plugins/wasi_nn/MLX/mlx/transformer.h [hunk 1]
```diff
@@ -46,10 +51,13 @@ class MultiHeadAttention : public Module {
     registerModule("out_proj", std::make_shared<Linear>(
                                    Linear(*ValueDims, *ValueOutputDims, Bias)));
   };
+
   mx::array forward(mx::array Queries, mx::array Keys, mx::array Values,
```
Comment (q82419) [inline plugins/wasi_nn/MLX/mlx/transformer.h:55]: @hydai 
This member function `WasmEdge::Host::WASINN::MLX::mlx::core::nn::MultiHeadAttention::forward` not implemented (neither override).
Is it OK?

**Review by hydai — DISMISSED** [review]

File: plugins/wasi_nn/MLX/mlx/transformer.h (hunk 1 above)
Comment (hydai) [reply plugins/wasi_nn/MLX/mlx/transformer.h:55]: cc @grorge123 
Please take a look.

Comment (grorge123) [reply plugins/wasi_nn/MLX/mlx/transformer.h:55]: In the current support model structure, most LLM models do not need the MultiHeadAttention::forward function. It is just one of the original MLX library provides functions.

**Review by hydai — APPROVED** [review]

**Pull request merged**